lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
```

Comment:
@colinsurprenant as the changes here as super easy and you want to work on fixing the close, please fix everything. This would be nice. Thanks